AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/MutationQueue/ProcessMutationErrorFromCloudOperationTests.swift

```diff
@@ -46,19 +52,719 @@ class ProcessMutationErrorFromCloudOperationTests: XCTestCase {
+        let completion: (Result<MutationEvent?, Error>) -> Void = { result in
+            guard case .success(let mutationEventOptional) = result else {
+                XCTFail("Should have been successful")
+                return
+            }
+            XCTAssertNil(mutationEventOptional)
+            expectCompletion.fulfill()
+        }
+
+        var eventListenerOptional: GraphQLOperation<MutationSync<AnyModel>>.EventListener?
+        let apiMutateCalled = expectation(description: "API was called")
+        mockAPIPlugin.responders[.mutateRequestListener] =
+            MutateRequestListenerResponder<MutationSync<AnyModel>> { request, eventListener in
+                guard let variables = request.variables, let input = variables["input"] as? [String: Any] else {
+                    XCTFail("The document variables property doesn't contain a valid input")
+                    return nil
+                }
+                XCTAssert(input["id"] as? String == localPost.id)
+                XCTAssert(request.document.contains("DeletePost"))
+                eventListenerOptional = eventListener
+                apiMutateCalled.fulfill()
+                return nil
+        }
+
+        let expectConflicthandlerCalled = expectation(description: "Expect conflict handler called")
+        let configuration = DataStoreConfiguration.custom(conflictHandler: { data, resolve  in
+            guard let localPost = data.local as? Post,
+                let remotePost = data.remote as? Post else {
+                XCTFail("Couldn't get Posts from local and remote data")
+                return
+            }
+
+            XCTAssertEqual(localPost.title, "localTitle")
+            XCTAssertEqual(remotePost.title, "remoteTitle")
+            expectConflicthandlerCalled.fulfill()
+            resolve(.retryLocal)
+        })
+
+        let operation = ProcessMutationErrorFromCloudOperation(dataStoreConfiguration: configuration,
+                                                               mutationEvent: mutationEvent,
+                                                               api: mockAPIPlugin,
+                                                               storageAdapter: storageAdapter,
+                                                               error: graphQLResponseError,
+                                                               completion: completion)
+
+        let queue = OperationQueue()
+        queue.addOperation(operation)
+
+        wait(for: [expectConflicthandlerCalled, apiMutateCalled], timeout: defaultAsyncWaitTimeout)
+        guard let eventListener = eventListenerOptional else {
+            XCTFail("Listener was not called through MockAPICategoryPlugin")
+            return
+        }
+        let updatedMetadata = MutationSyncMetadata(id: remotePost.id, deleted: true, lastChangedAt: 0, version: 3)
+        let local = MutationSync(model: try localPost.eraseToAnyModel(), syncMetadata: updatedMetadata)
```

Comment:
is there anything we can do to verify the data we are sending back through the event listener?  if i change deleted:false and the version to something like 99, the unit test still passes.

Reply:
this `local` model is just so we can pass something to the eventListener when we call it with .success(local) since we are mocking the API call to be successful. we don't really do anything with the service response after that, as long as it was successful, we finish the operation. it's the subscription reconcilation path that will take this mutation response and reconcile local store with it. a better name for this is probably `mockAPIResponse` or something and can be static data